### ISSUE TITLE
dotenv.config repetead in secret.ts and app.ts

### ISSUE THREAD

**Issue by JonathanLoscalzo** [microsoft/TypeScript-Node-Starter#132]
This is a question, maybe a silent bug. 

Why did we call `dotenv.config()` in both files?
It could be a _mistake_?

**Comment by wslyvh** [comment]
I think you're right. `dotenv.config()` should only be called once, preferably as earliest as possible.
App.ts is the most common approach, while Secret.ts seems more like an example for custom configurations.

**Comment by syffs** [comment]
@bowdenk7 It's been 10 months, could you please comment ? :smiley:

Also, I'm not 100% sure, but I think that `secret.ts` cannot work as it is:
- `dotenv` is smart enough to look for `.env` file at several directory levels
- working dir being `.../dist`, `fs.existsSync` is not that smart, and will not find `.env` file there

**Comment by bowdenk7** [comment]
and this is how we learned that I'm terrible at github notification management...

Thanks everyone for commenting - `app.ts` is the most common but in this case we pull it out to `secret.ts` to make it easier to put all of our environment variable management in one place. This makes it easier if we ever want to have multiple configs (imagine `.env.prod`, `.env.staging`, `.env.dev` - that sort of thing). Since whenever we import an environment value (which we already do [here](https://github.com/microsoft/TypeScript-Node-Starter/blob/master/src/app.ts#L13)) we load the `secret.ts` we no longer care how early we call it. Module resolution will always load it as soon as we need it.

I pushed this fix: [f14407dc](https://github.com/microsoft/TypeScript-Node-Starter/commit/f14407dc69f8904f3f8ec553cba7ce19064be9d4)
